### PR TITLE
[DataFactory] Add Integration Runtime Cmdlets.

### PR DESCRIPTION
**Pull request by zhangyd2015**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [ ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2/IntegrationRuntimes/GetAzureDataFactoryIntegrationRuntimeNodeIpCommand.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+using Microsoft.Azure.Management.DataFactory.Models;
+
+namespace Microsoft.Azure.Commands.DataFactoryV2
+{
+    [Cmdlet(
+        VerbsCommon.Get,
+        Constants.IntegrationRuntimeNodeIp,
+        DefaultParameterSetName = ParameterSetNames.ByIntegrationRuntimeName),
+        OutputType(typeof(IntegrationRuntimeNodeIpAddress))]
+    public class GetAzureDataFactoryIntegrationRuntimeNodeIpCommand : IntegrationRuntimeContextBaseCmdlet
```
Comment (markcowl) [inline src/ResourceManager/DataFactories/Commands.DataFactoryV2/IntegrationRuntimes/GetAzureDataFactoryIntegrationRuntimeNodeIpCommand.cs]: See my response in the review:  https://github.com/Azure/azure-powershell-pr/wiki/ADF-V2-Integration-Runtime-New-Cmdlets-(December-2017)-Notes#response 

If this is an expensive call, then you should provide an option switch on the existing Get-* cdlet to include this ifnormation, ratehr than havign a new cmdlet.

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2/IntegrationRuntimes/InvokeAzureDataFactoryIntegrationRuntimeUpgradeCommand.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        VerbsLifecycle.Invoke,
+        Constants.IntegrationRuntimeUpgrade,
+        DefaultParameterSetName = ParameterSetNames.ByIntegrationRuntimeName,
+        SupportsShouldProcess = true)]
+    public class InvokeAzureDataFactoryIntegrationRuntimeUpgradeCommand : IntegrationRuntimeCmdlet
+    {
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = Constants.HelpDontAskConfirmation)]
+        public SwitchParameter Force { get; set; }
```
Comment (markcowl) [inline src/ResourceManager/DataFactories/Commands.DataFactoryV2/IntegrationRuntimes/InvokeAzureDataFactoryIntegrationRuntimeUpgradeCommand.cs]: Do not include Force here, as you are doing thsi every time.  Instead, simply implement ShouldProcess

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2/IntegrationRuntimes/UpdateAzureDataFactoryIntegrationRuntimeNodeCommand.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = Constants.HelpIntegrationRuntimeJobsLimit)]
+        public int ConcurrentJobsLimit { get; set; }
+
+        [Parameter(
+            Mandatory = false, HelpMessage = Constants.HelpDontAskConfirmation)]
+        public SwitchParameter Force { get; set; }
```
Comment (markcowl) [inline src/ResourceManager/DataFactories/Commands.DataFactoryV2/IntegrationRuntimes/UpdateAzureDataFactoryIntegrationRuntimeNodeCommand.cs]: Force should not be necessary here

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2/packages.config [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.Azure.Management.DataFactory" version="0.2.1-preview" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.DataFactory" version="0.3.0-preview" targetFramework="net452" />
```
Comment (markcowl) [inline src/ResourceManager/DataFactories/Commands.DataFactoryV2/packages.config]: This needs to be cleaned up.  ActiveDirectory and AUthentication are not required.  If you use WindowsAzure.Storage and its depednecies and the other assemblies, then these need to be included in  the RequiredAssemblies in your psd1 file.  If you do not, then these should be removed form here.

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2/IntegrationRuntimes/InvokeAzureDataFactoryIntegrationRuntimeUpgradeCommand.cs (hunk 2 above)
Comment (zhangyd2015) [reply src/ResourceManager/DataFactories/Commands.DataFactoryV2/IntegrationRuntimes/InvokeAzureDataFactoryIntegrationRuntimeUpgradeCommand.cs]: Removed Force.

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2/IntegrationRuntimes/UpdateAzureDataFactoryIntegrationRuntimeNodeCommand.cs (hunk 3 above)
Comment (zhangyd2015) [reply src/ResourceManager/DataFactories/Commands.DataFactoryV2/IntegrationRuntimes/UpdateAzureDataFactoryIntegrationRuntimeNodeCommand.cs]: Removed.

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2/packages.config (hunk 4 above)
Comment (zhangyd2015) [reply src/ResourceManager/DataFactories/Commands.DataFactoryV2/packages.config]: Cleaned.

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2/IntegrationRuntimes/GetAzureDataFactoryIntegrationRuntimeNodeIpCommand.cs (hunk 1 above)
Comment (zhangyd2015) [reply src/ResourceManager/DataFactories/Commands.DataFactoryV2/IntegrationRuntimes/GetAzureDataFactoryIntegrationRuntimeNodeIpCommand.cs]: Changed it as your comments. Thanks.

**Comment by maddieclayton** [comment]
@zhangyd2015 Some merge conflicts have come up - can you please resolve these?

**Comment by markcowl** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/233/

**Pull request merged**